toy_problem/src/main.cpp

```diff
@@ -1,42 +1,130 @@
 #include "TaskAllocation.cpp"
+#include "yaml-cpp/yaml.h"
+#include <iostream>
+#include <fstream>
 
-int main()
+class TaskConfig
 {
+  public:
+    struct Agent
+    {
+      std::size_t id;
+      std::size_t wp;
+      std::size_t charging_wp;
+    };
+    
+    struct DeliveryTask
+    {
+      std::size_t id;
+      std::size_t pickup;
+      std::size_t dropoff;
+    };
+
+    double edge_length;
+    int grid_size;
+    std::vector<DeliveryTask> deliveries;
+    std::vector<Agent> agents;
+};
+
+TaskConfig load_task_config(const std::string& yaml_path)
+{
+  YAML::Node yaml_config = YAML::LoadFile(yaml_path);
+  TaskConfig config;
+  config.grid_size = yaml_config["grid_size"].as<int>();
+  config.edge_length = yaml_config["grid_length"].as<double>();
+
+  for ( auto task : yaml_config["tasks"]["delivery"])
+  {
+    TaskConfig::DeliveryTask delivery = {
+      task["id"].as<std::size_t>(),
+      task["pickup"].as<std::size_t>(),
+      task["dropoff"].as<std::size_t>()
+    };   
+    config.deliveries.push_back(delivery);
+  }
+
+  for ( auto ag: yaml_config["agents"])
+  {
+    TaskConfig::Agent agent = {
+      ag["id"].as<std::size_t>(),
+      ag["wp"].as<std::size_t>(),
+      ag["charging_wp"].as<std::size_t>()
+    };   
+    config.agents.push_back(agent);
+  }
+  return config;
+}
+
+void output_allocation_config(
+  const std::string& yaml_path,
+  const AssignedTasks& allocations )
+{
+  // allocations
+  
+  auto it = allocations.begin();
+
+  YAML::Emitter out;
+  out << YAML::Value << YAML::BeginSeq;
+  while(it != allocations.end())
+  {
+    out << YAML::BeginMap;
+    out << YAML::Key << "agent_id" << YAML::Value << it->first;
+    out << YAML::Key << "tasks";
+    out << YAML::Value << YAML::Flow << YAML::BeginSeq;
+    for( auto assignment : it->second)
+      out << assignment.task_id;
+    out << YAML::EndSeq << YAML::EndMap;
+    it++;
+  }
+  out << YAML::EndSeq;
+
+  std::ofstream fout(yaml_path);
+  fout << out.c_str();
+  fout.close();
+}
+
+//=========================================================================
+
+int main(int argc, char* argv[])
+{ 
   using SimpleMotionPowerSink = rmf_battery::agv::SimpleMotionPowerSink;
   using SimpleDevicePowerSink = rmf_battery::agv::SimpleDevicePowerSink;
-  srand(42);
-  // Graph
-  // 00-01-02-03
-  // |  |  |  |
-  // 04-05-06-07
-  // |  |  |  |
-  // 08-09-10-11
-  // |  |  |  | 
-  // 12-13-14-15
 
+  if (argc < 3)
+  {
+    std::cerr<< " Incorrect number of args, input format: \n"
+             << " \t ./toy_problem {INPUT_YAML} {OUTPUT_YAML}" << std::endl;
+    return 0;
+  }
+
+  // load yaml file
+  std::string task_config(argv[argc-2]);
```

Comment:
Nitpick but readability is better with absolute indices 
```suggestion
  std::string task_config(argv[1]);
```